src/main/java/io/boomerang/eventing/nats/jetstream/PubTransmitter.java

```diff
@@ -80,32 +87,114 @@ public void publish(String subject, String message)
     Connection connection = connectionPrimer.getActiveConnection();
-
     if (connection == null) {
+      storeMessage(subject, message);
       throw new NoNatsConnectionException("No connection to the NATS server!");
     }
 
     // Get Jetstream stream from the NATS server (this will also automatically create the stream if
     // not present on the server)
+    try {
+      getJetStream(connection);
+    } catch (NullPointerException npe) {
+      storeMessage(subject, message);
+    }
+
+    // Create the NATS message
+    Message natsMessage = createNATSMessage(subject, message);
+
+    // Publish the message
+    PublishAck publishAck = connection.jetStream().publish(natsMessage);
+
+    System.out.println("Message \"" + message + "\" with subject \"" + subject + "\" published to stream");
+    printObject(publishAck);
+    // logger.debug("Message published to the stream! " + publishAck);
+  }
+
+  private void getJetStream(Connection connection) throws IOException, JetStreamApiException {
     StreamInfo streamInfo = StreamManager.getStreamInfo(connection, streamConfiguration,
         pubOnlyConfiguration.isAutomaticallyCreateStream());
 
     if (streamInfo == null) {
       throw new StreamNotFoundException("Stream could not be found! Consider enabling "
           + "`automaticallyCreateStream` in `PubOnlyConfiguration`");
     }
+  }
 
-    // Create the NATS message
-    // @formatter:off
-    Message natsMessage = NatsMessage.builder()
-        .subject(subject)
-        .data(message, StandardCharsets.UTF_8)
-        .build();
-    // @formatter:on
+  // TODO replace this w one line where needed instead of a method call?
+  private void storeMessage(String subject, String message) {
+    System.out.println("Storing message to send later");
+    failedMessages.add(createNATSMessage(subject, message));
+    printObject(failedMessages);
+  }
 
-    // Publish the message
-    PublishAck publishAck = connection.jetStream().publish(natsMessage);
+  // TODO replace this w one line where needed instead of a method call?
+  private NatsMessage createNATSMessage(String subject, String message) {
+    return NatsMessage.builder().subject(subject).data(message, StandardCharsets.UTF_8).build();
+  }
+
+  private void publishFailedMessages() {
+    // removed unnecessary arguments from this method
+    // convert failedMessages List<Message> -> ConcurrentLinkedQueue<Message>
+    // remove one message at a time after successful publish
+    // attempt actions inside synchronized while loop,
+    // break if queue is empty or if connection fails again
+    
+    while (true) {
+      synchronized (this) {
+        try {
+          System.out.println("Resending failed message");
+          PublishAck publishAck =
+              connectionPrimer.getActiveConnection().jetStream().publish(failedMessages.peek());
+          System.out.println("PublishAck:");
+          printObject(publishAck);
+          if (publishAck != null) {
```

Comment:
`PublishAck` is not an indicator whether the message was published successfully or not. You must check for raised exceptions from the `publish` method. See the javadoc for this method for more information.